packages/slider/src/Slider.ts

```diff
@@ -348,6 +357,60 @@ export class Slider extends SizedMixin(ObserveSlotText(SliderHandle, ''), {
         `;
     }
 
+    private _cachedValue: number | undefined;
+
+    /**
+     * @description calculates the fill width
+     * @param fillStartValue
+     * @param currentValue
+     * @param cachedValue
+     * @returns
+     */
+    private getOffsetWidth(
+        fillStartValue: number,
+        currentValue: number,
+        cachedValue: number
+    ): number {
+        const distance =
+            fillStartValue === cachedValue
+                ? Math.abs(currentValue - fillStartValue)
+                : Math.abs(
+                      Math.min(currentValue, cachedValue) - fillStartValue
+                  );
+
+        return (distance / (this.max - this.min)) * 100;
+    }
+
+    private getOffsetPosition(v: number): number {
+        const val = ((v - this.min) / (this.max - this.min)) * 100;
+        return val;
+    }
+
+    private renderFillOffset(): TemplateResult {
+        if (!this.fillStart || !this.fillStartPoint || !this._cachedValue) {
+            return html``;
+        }
+        return html`
+            <div
+                class=${`fill ${
+                    !!(this.value > this.fillStartPoint) ? 'offset' : ''
+                }`}
```

Comment:
using `classMap({fill: true, offset: this.value > this.fillStartPoint})` might make this easier to read.